core/trino-parser/src/main/java/io/trino/sql/tree/MergeDelete.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class MergeDelete
+        extends MergeCase
+{
+    public MergeDelete(NodeLocation location, Optional<Expression> expression)
+    {
+        super(Optional.of(location), expression);
+    }
+
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context)
+    {
+        return visitor.visitMergeDelete(this, context);
+    }
+
+    @Override
+    public List<String> getSetColumns()
+    {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public List<Expression> getSetExpressions()
+    {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public List<? extends Node> getChildren()
+    {
+        return expression.isPresent() ? ImmutableList.of(expression.get()) : ImmutableList.of();
```

Comment:
This can be
```java
return expression.map(ImmutableList::of).orElseGet(ImmutableList::of);
```
If the parser module wasn't limited to JDK 8, we could use
```java
return expression.stream().collect(toImmutableList());
```

Reply:
Yes, much nicer.  Changed.